tests/scattering1d/test_timefrequency_scattering_tensorflow.py

```diff
@@ -0,0 +1,478 @@
+import ssl
+import numpy as np
+import pytest
+import tensorflow as tf
+
+import kymatio
+from kymatio.scattering1d.core.scattering1d import scattering1d
+from kymatio.scattering1d.filter_bank import compute_temporal_support, gauss_1d
+from kymatio.scattering1d.core.timefrequency_scattering import (
+    joint_timefrequency_scattering,
+    time_scattering_widthfirst,
+    frequency_scattering,
+    time_averaging,
+    frequency_averaging,
+)
+from kymatio.scattering1d.frontend.base_frontend import TimeFrequencyScatteringBase
+from kymatio.scattering1d.frontend.numpy_frontend import TimeFrequencyScatteringNumPy
+from kymatio.scattering1d.frontend.tensorflow_frontend import TimeFrequencyScatteringTensorFlow
+
+
+
+def test_jtfs_build():
+    # Test __init__
+    jtfs = TimeFrequencyScatteringBase(J=10, J_fr=3, shape=4096, Q=8, backend="tensorflow")
+    assert jtfs.F is None
+
+    # Test Q_fr
+    jtfs.build()
+    assert jtfs.Q_fr == (1,)
+
+    # Test parse_T(self.F, self.J_fr, N_input_fr, T_alias='F')
+    assert jtfs.F == (2**jtfs.J_fr)
+    assert jtfs.log2_F == jtfs.J_fr
+
+    # Test that frequential padding is sufficient
+    N_input_fr = len(jtfs.psi1_f)
+    assert (jtfs._N_padded_fr - N_input_fr) > (2 * jtfs.F)
+    assert (jtfs._N_padded_fr - N_input_fr) > (2 * (2**jtfs.J_fr))
+    phi_f = gauss_1d(2 * jtfs._N_padded_fr, jtfs.sigma0 / jtfs.F)
+    h_double_support = np.fft.fftshift(np.fft.ifft(phi_f))
+    h_current_support = h_double_support[
+        (jtfs._N_padded_fr // 2) : -jtfs._N_padded_fr // 2
+    ]
+    l1_current = np.linalg.norm(h_current_support, ord=1)
+    l1_double = np.linalg.norm(h_double_support, ord=1)
+    l1_residual = l1_double - l1_current
+    criterion_amplitude = 1e-3
+    assert l1_residual < criterion_amplitude
+
+    # Test that padded frequency domain is divisible by max subsampling factor
+    assert (jtfs._N_padded_fr % (2**jtfs.J_fr)) == 0
+
+
+
+def test_Q():
+    jtfs_kwargs = dict(J=10, J_fr=3, shape=4096, Q=8, backend="tensorflow")
+    # test different cases for Q_fr
+    with pytest.raises(ValueError) as ve:
+        TimeFrequencyScatteringBase(Q_fr=0.9, **jtfs_kwargs).build()
+    assert "Q_fr must be >= 1" in ve.value.args[0]
+
+    # test different cases for Q_fr
+    with pytest.raises(ValueError) as ve:
+        TimeFrequencyScatteringBase(Q_fr=[1], **jtfs_kwargs).build()
+    assert "Q_fr must be an integer or 1-tuple" in ve.value.args[0]
+
+    with pytest.raises(NotImplementedError) as ve:
+        TimeFrequencyScatteringBase(Q_fr=(1, 2), **jtfs_kwargs).build()
+    assert "Q_fr must be an integer or 1-tuple" in ve.value.args[0]
+
+
+def test_check_runtime_args():
+    kwargs = dict(J=10, J_fr=3, shape=4096, Q=8)
+    S = TimeFrequencyScatteringBase(**kwargs)
```

Comment:
why are we testing this here? i don't see tensorflow frontend and this is duplicate code?